tensornetwork/block_sparse/charge_test.py

```diff
@@ -42,3 +42,238 @@ def test_BaseCharge_copy():
   assert Q.unique_charges is not Qcopy.unique_charges
   np.testing.assert_allclose(Q.charge_labels, Qcopy.charge_labels)
   np.testing.assert_allclose(Q.unique_charges, Qcopy.unique_charges)
+
+
+def test_intersect_1():
+  a = np.array([[0, 1, 2], [2, 3, 4]])
+  b = np.array([[0, -2, 6], [2, 3, 4]])
+  out = intersect(a, b, axis=1)
+  np.testing.assert_allclose(np.array([[0], [2]]), out)
+
+
+def test_intersect_2():
+  a = np.array([[0, 1, 2], [2, 3, 4]])
+  b = np.array([[0, -2, 6, 2], [2, 3, 4, 4]])
+  out, la, lb = intersect(a, b, axis=1, return_indices=True)
+  np.testing.assert_allclose(np.array([[0, 2], [2, 4]]), out)
+  np.testing.assert_allclose(la, [0, 2])
+  np.testing.assert_allclose(lb, [0, 3])
+
+
+def test_intersect_3():
+  a = np.array([0, 1, 2, 3, 4])
+  b = np.array([0, -1, 4])
+  out = intersect(a, b)
+  np.testing.assert_allclose([0, 4], out)
+
+
+def test_intersect_3():
+  a = np.array([0, 1, 2, 3, 4])
+  b = np.array([0, -1, 4])
+  out, la, lb = intersect(a, b, return_indices=True)
+  np.testing.assert_allclose([0, 4], out)
+  np.testing.assert_allclose(la, [0, 4])
+  np.testing.assert_allclose(lb, [0, 2])
+
+
+def test_fuse_degeneracies():
+  d1 = np.asarray([0, 1])
+  d2 = np.asarray([2, 3, 4])
+  fused_degeneracies = fuse_degeneracies(d1, d2)
+  np.testing.assert_allclose(fused_degeneracies, np.kron(d1, d2))
+
+
+def test_U1Charge_charges():
+  D = 100
+  B = 6
+  charges = np.random.randint(-B // 2, B // 2 + 1, D).astype(np.int16)
+
+  q1 = U1Charge(charges)
+  assert np.all(q1.charges == charges)
+
+
+def test_U1Charge_dual():
+  D = 100
+  B = 6
+  charges = np.random.randint(-B // 2, B // 2 + 1, D).astype(np.int16)
+
+  q1 = U1Charge(charges)
+  assert np.all(q1.dual(True).charges == -charges)
+
+
+def test_U1Charge_fusion():
+
+  def run_test():
+    D = 2000
+    B = 6
+    O1 = np.random.randint(-B // 2, B // 2 + 1, D).astype(np.int16)
+    O2 = np.random.randint(-B // 2, B // 2 + 1, D).astype(np.int16)
+    P1 = np.random.randint(0, B + 1, D).astype(np.int16)
+    P2 = np.random.randint(-B // 2, B // 2 + 1, D).astype(np.int16)
+    Q1 = np.random.randint(1, B + 1, D).astype(np.int16)
+    Q2 = np.random.randint(1, B + 1, D).astype(np.int16)
```

Comment:
In general it's a good idea to avoid random values in a test if you can help it. Non-seeded random values tend to cause tests to be flakey. Imagine having a test fail and then not be able to fix it because it depended on a specific set of random inputs.

I'll let it slide for now but keep it in mind to prefer to hard code things

Reply:
no problem, I'll fix this by seeding